app/controllers/admin/merchants_controller.rb

```diff
@@ -14,6 +14,11 @@ def update
       merchant.deactivate_items
       flash[:notice] = "#{merchant.name} is now disabled."
       redirect_to admin_merchants_path
+    else  
+      merchant.update(enabled?: true)
+      merchant.activate_items
+      flash[:notice] = "#{merchant.name} is now enabled."
+      redirect_to admin_merchants_path
```

Comment:
This code looks great and really clean!

Reply:
looks great maybe helper method for a lot of that if logic, to shorten the controller action and follow single responsibility